rmw_opendds_cpp/src/rmw_publisher.cpp

```diff
@@ -110,20 +111,20 @@ rmw_create_publisher(
   }
   std::string type_name = _create_type_name(callbacks, "msg");
   // Past this point, a failure results in unrolling code in the goto fail block.
-  //DDS::TypeCode * type_code = nullptr;
   DDS::DataWriterQos datawriter_qos;
   DDS::PublisherQos publisher_qos;
   DDS::ReturnCode_t status;
-  DDS::Publisher * dds_publisher = nullptr;
-  DDS::DataWriter * topic_writer = nullptr;
-  DDS::Topic * topic = nullptr;
-  DDS::TopicDescription * topic_description = nullptr;
+  DDS::Publisher_var dds_publisher;
+  DDS::DataWriter_var topic_writer;
+  DDS::Topic_var topic;
+  DDS::TopicDescription_var topic_description;
   void * info_buf = nullptr;
   void * listener_buf = nullptr;
   OpenDDSPublisherListener * publisher_listener = nullptr;
   OpenDDSStaticPublisherInfo * publisher_info = nullptr;
   rmw_publisher_t * publisher = nullptr;
   std::string mangled_name = "";
+  OpenDDSStaticSerializedDataTypeSupport_var ts = new OpenDDSStaticSerializedDataTypeSupportImpl();
```

Comment:
Recognizing that we're focused on construction and not deconstruction at this point in the project, is the `delete` something we need to do or is this a smart pointer (or similar) type?
```suggestion
//TODO add delete
  OpenDDSStaticSerializedDataTypeSupport_var ts = new OpenDDSStaticSerializedDataTypeSupportImpl();
```

Reply:
`_var` is a smart pointer.